python/lsst/pipe/tasks/hips.py

```diff
@@ -604,3 +614,907 @@ def build_quantum_graph(
+        r_term.lambda_max = 691.
+        self.spectral_ranges['r'] = r_term
+        i_term = HipsPropertySpectralTerm()
+        i_term.lambda_min = 691.
+        i_term.lambda_max = 818.
+        self.spectral_ranges['i'] = i_term
+        z_term = HipsPropertySpectralTerm()
+        z_term.lambda_min = 818.
+        z_term.lambda_max = 922.
+        self.spectral_ranges['z'] = z_term
+        y_term = HipsPropertySpectralTerm()
+        y_term.lambda_min = 970.
+        y_term.lambda_max = 1060.
+        self.spectral_ranges['y'] = y_term
+
+
+class GenerateHipsConnections(pipeBase.PipelineTaskConnections,
+                              dimensions=("instrument", "band"),
+                              defaultTemplates={"coaddName": "deep"}):
+    hips_exposure_handles = pipeBase.connectionTypes.Input(
+        doc="HiPS-compatible HPX images.",
+        name="{coaddName}Coadd_hpx",
+        storageClass="ExposureF",
+        dimensions=("healpix11", "band"),
+        multiple=True,
+        deferLoad=True,
+    )
+
+
+class GenerateHipsConfig(pipeBase.PipelineTaskConfig,
+                         pipelineConnections=GenerateHipsConnections):
+    """Configuration parameters for GenerateHipsTask."""
+    # WARNING: In general PipelineTasks are not allowed to do any outputs
+    # outside of the butler.  This task has been given (temporary)
+    # Special Dispensation because of the nature of HiPS outputs until
+    # a more controlled solution can be found.
+    hips_base_uri = pexConfig.Field(
+        doc="URI to HiPS base for output.",
+        dtype=str,
+        optional=False,
+    )
+    min_order = pexConfig.Field(
+        doc="Minimum healpix order for HiPS tree.",
+        dtype=int,
+        default=3,
+    )
+    properties = pexConfig.ConfigField(
+        dtype=HipsPropertyConfig,
+        doc="Configuration for property file.",
+    )
+    png_gray_asinh_minimum = pexConfig.Field(
+        doc="AsinhMapping intensity to be mapped to black for grayscale png scaling.",
+        dtype=float,
+        default=0.0,
+    )
+    png_gray_asinh_stretch = pexConfig.Field(
+        doc="AsinhMapping linear stretch for grayscale png scaling.",
+        dtype=float,
+        default=2.0,
+    )
+    png_gray_asinh_softening = pexConfig.Field(
+        doc="AsinhMapping softening parameter (Q) for grayscale png scaling.",
+        dtype=float,
+        default=8.0,
+    )
+
+
+class GenerateHipsTask(pipeBase.PipelineTask):
+    """Task for making a HiPS tree with fits and grayscale pngs."""
+    ConfigClass = GenerateHipsConfig
+    _DefaultName = "generateHips"
+
+    @timeMethod
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        dims = inputRefs.hips_exposure_handles[0].dataId.names
+        order = None
+        for dim in dims:
+            if "healpix" in dim:
+                order = int(dim.split("healpix")[1])
+                healpix_dim = dim
+        if order is None:
```

Comment:
If the loop does `break` this can be an `else`.

Reply:
I did not know that.  Huh!

Reply:
Wait, this is the opposite ... it is supposed to break.

Reply:
That's my point. If you break out the else won't trigger. `else` only runs if the loop completes.